packages/nextjs/pages/_app.tsx

```diff
@@ -25,37 +32,30 @@ const ScaffoldEthApp = ({ Component, pageProps }: AppProps) => {
   }, [setNativeCurrencyPrice, price]);
 
   return (
-    <>
+    <RainbowKitProvider
+      chains={appChains.chains}
+      avatar={BlockieAvatar}
+      theme={mounted ? (isDarkMode ? darkTheme() : lightTheme()) : lightTheme()}
+    >
       <div className="flex min-h-screen flex-col">
         <main className="relative flex flex-1 flex-col">
           <Component {...pageProps} />
         </main>
       </div>
       <Toaster />
-    </>
+    </RainbowKitProvider>
   );
 };
 
 const ScaffoldEthAppWithProviders = (props: AppProps) => {
-  // This variable is required for initial client side rendering of correct theme for RainbowKit
-  const [isDarkTheme, setIsDarkTheme] = useState(true);
-  const { isDarkMode } = useDarkMode();
-  useEffect(() => {
-    setIsDarkTheme(isDarkMode);
-  }, [isDarkMode]);
-
   return (
     <PlausibleProvider domain="abi.ninja">
-      <WagmiConfig config={wagmiConfig}>
-        <NextNProgress />
-        <RainbowKitProvider
-          chains={appChains.chains}
-          avatar={BlockieAvatar}
-          theme={isDarkTheme ? darkTheme() : lightTheme()}
-        >
+      <ThemeProvider>
```

Comment:
Maybe we pass `enableSystem` : `<ThemeProvider enableSystem>` so that by default when the user first-time land on abi.ninja we use his system theme by default

Reply:
Ohh my bad, yup you are right its true by default